### PR TITLE
Add Args.maxShrinks

### PR DESCRIPTION
**Pull request by phadej**
cc @thumphries

### CODE REVIEW

File: Test/QuickCheck/Test.hs [hunk 1]
```diff
@@ -371,6 +375,10 @@ foundFailure st res ts =
 localMin :: State -> P.Result -> P.Result -> [Rose P.Result] -> IO (Int, Int, Int)
 localMin st MkResult{P.theException = Just e} lastRes _
   | isInterrupt e = localMinFound st lastRes
+-- Don't try to shrink for too long
+localMin st res _ ts
+  | numSuccessShrinks st + numTotTryShrinks st >= numTotMaxShrinks st =
```
Comment (thumphries) [inline Test/QuickCheck/Test.hs]: Could `numSuccessShrinks st + numTotTryShrinks st` overflow?

**Comment by thumphries** [comment]
Thanks for implementing this!

👍

File: Test/QuickCheck/Test.hs (hunk 1 above)
Comment (phadej) [inline Test/QuickCheck/Test.hs]: I don't think so, as the check is done on every iteration.

**Comment by nick8325** [comment]
Using `Test.QuickCheck.Function` can result in a very large number of shrink steps (sometimes into the thousands), and if you don't run shrinking to completion you get a function which is not showable. So I would rather have the default be unlimited shrinking, and the user can explicitly set `maxShrinks` if they want less.

**Comment by nick8325** [comment]
P.S. thanks for the patch!

**Comment by phadej** [comment]
amended

**Comment by nick8325** [comment]
Huh, Travis fails now. Looks like `prop_B1` needs to be updated? I think instead of `$quickCheckAll` you want (IIRC) `$forAllProperties (quickCheckWithResult stdArgs { maxShrinks = 10000 })`.

**Comment by phadej** [comment]
amended once again

**Pull request merged**